railgun/test/integration/suite_test.go

```diff
@@ -325,6 +328,13 @@ func waitForExistingClusterReadiness(ctx context.Context, cluster ktfkind.Cluste
 						ready <- ktfkind.ProxyReadinessEvent{Err: err}
 						break
 					}
+					proxyHTTPSURL, err = url.Parse(fmt.Sprintf("https://%s:%d", svc.Status.LoadBalancer.Ingress[0].IP, 443))
+					if err != nil {
+						ready <- ktfkind.ProxyReadinessEvent{Err: err}
+						break
+					}
+					addr := net.ParseIP(svc.Status.LoadBalancer.Ingress[0].IP)
+					proxyIP = &addr
```

Comment:
It would be better validating the IP address since the status changes.

Reply:
It's basically copying what the others do, and they don't validate when populating URLs, only blindly format the URL strings.

@shaneutt I don't often use this particular piece of functionality, but I believe we're just relying on the check above (`len(svc.Status.LoadBalancer.Ingress) == 1`) to guarantee that _any_ status info is present, and assuming that it must then have a valid IP, correct? If not, this entire block is bypassed, the proxy URL variable isn't set, and the `proxyURL != nil` condition after fails, continuing the loop, correct? Is there any case where the status length check would succeed without actually having an IP available? Would we need to tighten this?

Reply:
Do you mind I chime in since Shane already on the beach with beer :). 
 technically Line 336 should return a valid IP if no error/exception, since ProxyURL already present, an easier way retrieving proxy IP
proxyURL.Hostname()

Reply:
```suggestion
					proxyIP = proxyURL.Hostname()
```

Reply:
This actually broke because the types aren't the same :facepalm: 

We could parse  `proxyURL.Hostname()` to the dedicated IP type, but thinking over it some more, we could potentially switch to DNS hostnames for the URLs in the future, and if so that'd break parsing. The IP field in status is guaranteed to be an IP string, so using that is safer.

Reply:
If work in the cloud env, HostName(), or the external IP of the Proxy ( Loadbalancer Type) is a valid DNS which is accessible publicly.

Are you still need an IP address in that case ?

Reply:
In the context it's used, yes. It's for things like https://github.com/Kong/kubernetes-ingress-controller/pull/1439/files#diff-3b890c93c344e46a0243585a4ab4a5b3d9c8eba953fc1ad9693ceaa81c70746aR536-R547 where a test needs to override DNS and pretend that some other hostname resolves to the proxy IP.